rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java

```diff
@@ -776,15 +776,38 @@ public static byte[] serializeCoinbaseInformation(CoinbaseInformation coinbaseIn
     }
 
     public static byte[] serializeSha256Hash(Sha256Hash hash) {
+        if (hash == null){
+            return new byte[]{};
+        }
         return RLP.encodeElement(hash.getBytes());
     }
 
-    public static Sha256Hash deserializeSha256Hash(byte[] data) {
+    public static Optional<Sha256Hash> deserializeSha256Hash(byte[] data) {
         RLPElement element = RLP.decodeFirstElement(data, 0);
         if (element == null) {
-            return null;
+            return Optional.empty();
+        }
+        return Optional.of(Sha256Hash.wrap(element.getRLPData()));
+    }
+
+    public static Optional<Keccak256> deserializeKeccak256(byte[] data) {
+        if (data == null) {
+            return Optional.empty();
+        }
+
+        RLPList rlpList = (RLPList) RLP.decode2(data).get(0);
+        if (rlpList.size() != 1) {
+            throw new RuntimeException(String.format("Invalid serialized keccak256. Expected 1 element, but got %d", rlpList.size()));
+        }
+
+        return Optional.of(new Keccak256(rlpList.get(0).getRLPRawData()));
+    }
+
+    public static byte[] serializeKeccak256(Keccak256 data) {
+        if (data == null){
+            return new byte[]{};
         }
-        return Sha256Hash.wrap(element.getRLPData());
+        return RLP.encodeList(RLP.encodeElement(data.getBytes()));
```

Comment:
I thought it was the correct way to do it. I left a question in the channel regarding this and the answer was to take a look at one of the existing methods which were using that approach to serialize/deserialize the value.

However, I did some tests changing it to use only the element encoding and it worked. So, I refactored the code and now it's being treated as a single element, as it really is, instead of as a list.

Reply:
Cool